custom_components/tuya_local/devices/kogan_lx10_vacuum.yaml

```diff
@@ -0,0 +1,259 @@
+name: Kogan LX10 Robot Vacuum
+products:
+  - id: ldb9ygikft5awhlu
+    name: Kogan LX10 Robot Vacuum
+primary_entity:
+  entity: vacuum
+  dps:
+    - id: 101
+      name: clean_switch
+      type: boolean
+    - id: 102
+      name: pause_switch
+      type: boolean
+    - id: 103
+      name: charge_switch
+      type: boolean
+    - id: 104
+      name: clean_mode
+      type: string
+      mapping:
+        - dps_val: smart
+          value: smart
+        - dps_val: pose
+          value: pose
+        - dps_val: zone
+          value: zone
+        - dps_val: backcharge
+          value: backcharge
+    - id: 105
+      name: robot_state
+      type: string
+      mapping:
+        - dps_val: idle
+          value: idle
+        - dps_val: pointing\areaing
+          value: pointing\areaing
+        - dps_val: totaling
+          value: totaling
+        - dps_val: sweep
+          value: sweep
+        - dps_val: mop
+          value: mop
+        - dps_val: fault
+          value: fault
+        - dps_val: pause
+          value: pause
+        - dps_val: chargring
+          value: chargring
+        - dps_val: tocharge
+          value: tocharge
```

Comment:
Change this value to `returning_to_base` to have it recognised by HA as a special status that will affect the state.